### PR TITLE
feat: (sfui2-594) display content

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue

https://vsf.atlassian.net/browse/SFUI2-594?atlOrigin=eyJpIjoiMjE2N2RjMmUzZGJlNGM4ZGE3OWQwNmM5YzkzMDFkM2QiLCJwIjoiaiJ9

# Scope of work

- extracted text content of hero and display to DisplayContent component

# Screenshots of visual changes

# Checklist

- [ ] Self code-reviewed
- [ ] Changes documented
- [ ] Semantic HTML
- [ ] SSR-friendly
- [ ] Caching friendly
- [ ] a11y for WCAG 2.0 AA
- [ ] examples created
- [ ] created cypress tests

### CODE REVIEW

File: packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import classNames from 'classnames';
+import { VsfDisplayContentProps } from './types';
+
+export default function Display({ children, title, subtitle, description, className }: VsfDisplayContentProps) {
+  return (
+    <div className={classNames('vsf-display-content', className)} data-testid="display-content">
+      <p className="vsf-display-content__subtitle">{subtitle}</p>
+      <h1 className="vsf-display-content__title">{title}</h1>
```
Comment (justyna-13) [inline packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx:8]: I wonder about h1 tag in this component. I think we can't assume that a user will have only one display content per page so there is quite big probability there will be more than one h1 tags. What do you think about possibility to choose number of a heading from a prop? I know there wasn't such proposition in api but maybe it's worth considering? @FRSgit

**Review by justyna-13 — COMMENTED** [review]
Looks great!

File: packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx (hunk 1 above)
Comment (aniamusial) [reply packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx:8]: Or a  separate component to render texts? @FRSgit

Comment (aniamusial) [reply packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx:8]: I think rendering dynamic tags may not be this easy from react perspective right @dkacper ?

Comment (justyna-13) [reply packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx:8]: Lovely react :D

Comment (FRSgit) [reply packages/sfui/frameworks/react/components/VsfDisplayContent/VsfDisplayContent.tsx:8]: We have talked about that with Kacper and Simon, the conclusion is:

Let's add prop for tag customization: `titleTag` (for vue) and `titleAs` (for react)

File: apps/preview/next/pages/examples/VsfDisplayContent.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @next/next/no-img-element */
+// Disabled due to issue with placement of the Next Image
```
Comment (FRSgit) [inline apps/preview/next/pages/examples/VsfDisplayContent.tsx:2]: Suggestion: This probably is not needed

File: apps/docs/components/components/displaycontent.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+| className       | `string`                |               |                                   |
+| children        | `ReactNode`             |               |                                   |
+<!-- end react -->
+
+<!-- vue -->
+## Slots
+
+| Slot name | Description         |
+| --------- | ------------------- |
+| default   | Content below text content |
```
Comment (FRSgit) [inline apps/docs/components/components/displaycontent.md:29]: Suggestion: we support `ReactNode`s in React, so we should support slots for `title`, `subtitle` and `description` as well. `ReactNode` usage and slots are counterparts

Comment (FRSgit) [reply apps/docs/components/components/displaycontent.md:29]: Let's keep it simple and not support `ReactNode` for all these props whatsoever.

We'll add it based on user feedback

**Review by AdamPawlinski — COMMENTED** [review]
Looks good, but one question - is a separate example for this component needed? I think this component won't be used outside VsfDisplay so maybe we can skip it, wdyt?

File: apps/docs/components/components/displaycontent.md (hunk 3 above)
Comment (aniamusial) [reply apps/docs/components/components/displaycontent.md:29]: Ok, its just something that was added in tasks requirements, thats why its here :D

**Comment by aniamusial** [comment]
> Looks good, but one question - is a separate example for this component needed? I think this component won't be used outside VsfDisplay so maybe we can skip it, wdyt?

Its also used in Hero, according to @FRSgit we should have a separate example and docs for it

File: packages/sfui/frameworks/sass/components/VsfDisplayContent.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
+.vsf-display-content {
+  @apply p-4 md:p-10 md:flex md:flex-col md:justify-center md:items-start md:basis-2/4;
```
Comment (FRSgit) [inline packages/sfui/frameworks/sass/components/VsfDisplayContent.scss:2]: Suggestion: That styling should be still a part of `VsfDisplay` component

Comment (aniamusial) [reply packages/sfui/frameworks/sass/components/VsfDisplayContent.scss:2]: @FRSgit It is exactly the same for both - hero and display, should I still move it back to these components?

File: packages/sfui/frameworks/react/components/VsfHero/VsfHero.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ export default function Hero({
     <div style={variables} className={classNames('vsf-hero', className)} data-testid="hero">
       <div className={classNames('vsf-hero__container', `vsf-hero__container--${direction}`)}>
         <div className="vsf-hero__media">{slotMedia}</div>
-        <div className="vsf-hero__content">{children}</div>
```
Comment (FRSgit) [inline packages/sfui/frameworks/react/components/VsfHero/VsfHero.tsx:23]: Suggestion: Same here, I believe that the HTML structure & CSS code needed for alignment of content within hero component should stay in Hero component code.
For example - we want to strictly specify that the space for content is 50% of the parent width.

**Comment by AdamPawlinski** [comment]
> > Looks good, but one question - is a separate example for this component needed? I think this component won't be used outside VsfDisplay so maybe we can skip it, wdyt?
> 
> Its also used in Hero, according to @FRSgit we should have a separate example and docs for it

Ok, thanks for explanation :)

**Review by AdamPawlinski — APPROVED** [review]

**Pull request merged**